webservice/src/main/java/com/hortonworks/registries/webservice/RegistryApplication.java

```diff
@@ -94,23 +94,24 @@ public void run(RegistryConfiguration registryConfiguration, Environment environ
     }
 
     private void initializeUGI(RegistryConfiguration conf) throws IOException {
-        LOG.debug("Initialization of User Group ininformation...");
+        LOG.debug("Initialization of User Group information...");
         if (UserGroupInformation.isSecurityEnabled()) {
             LOG.debug("UGI.isSecurityEnabled() = true.");
 
-            AuthMethodConfiguration authMethodConf = conf.getAuthenticationMethod();
+            ServiceAuthenticationConfiguration authMethodConf = conf.getServiceAuthenticationConfiguration();
             if(authMethodConf != null) {
-                String serverPrincipal = authMethodConf.getServerPrinciple();
-                String keyTab = authMethodConf.getServerPrincipleKeytab();
+                Map<String,String> serviceAuthenticationProperties = authMethodConf.getProperties();
+                String principal = serviceAuthenticationProperties.get("principal");
+                String keytab = serviceAuthenticationProperties.get("keytab");
```

Comment:
You should validate the principal and keytab for null and empty.

You should also catch exception from below, log it as error and then fallback to simple.

> UserGroupInformation.loginUserFromKeytab(principal, keytab);